server/src/main/java/org/elasticsearch/index/engine/CombinedDeletionPolicy.java

```diff
@@ -38,6 +40,17 @@ public class CombinedDeletionPolicy extends IndexDeletionPolicy {
     private final SoftDeletesPolicy softDeletesPolicy;
     private final LongSupplier globalCheckpointSupplier;
     private final Map<IndexCommit, Integer> snapshottedCommits; // Number of snapshots held against each commit point.
+
+    interface CommitsListener {
+
+        void onNewAcquiredCommit(IndexCommit commit);
+
+        void onDeletedCommit(IndexCommit commit);
```

Comment:
This is maybe a bit premature as I'm not 100% this is the hook we'll use to detect commit point deletions. But it's handful for testing that released commit points are effectively deleted, so I left it here.